lighthouse-core/lib/network-request.js

```diff
@@ -453,6 +453,24 @@ class NetworkRequest {
     return requestId.replace(/(:redirect)+$/, '');
   }
 
+  /**
+   * Return the body of the response with the given ID. Rejects if getting the
+   * body times out.
+   * @param {LH.Gatherer.FRProtocolSession} session
+   * @param {string} requestId
+   * @param {number} [timeout]
+   * @return {Promise<string>}
+   */
+  static async getRequestContent(session, requestId, timeout = 1000) {
```

Comment:
while we're in the reorg mode...WDYT about `fetchResponseBodyFromCache`?

- `fetch` to signal that it's an operation that takes some time over the protcol, not just a getter
- `FromCache` to signify that it's not re-requesting anything, it's just trying to get it from memory cache (`FromMemory` or `FromMemoryCache` alternates?)
- `ResponseBody` because it is weird that we say we're getting the request content when it's the *response* content, request is sort of implicit in all of these operations

Reply:
maybe `fetch` should be reserved for things fetched over the network, though?

Reply:
> maybe fetch should be reserved for things fetched over the network, though?

That was the point of `FromCache`, an alternate suggestion? `fetchFromCache` has a weaker "over the network" connection to me than `get`'s connection to "returns immediately"

Reply:
I guess I always found the name descriptive enough except for the `request` part you mentioned, so `getResponseContent` seems ok to me :)

If it was on `NetworkRequest` I could definitely see the issue with `get` sounding like just a local variable, but between the `async` and being in a `driver/*` file, I'd just assume it's doing something over the protocol, but maybe that's just because I already know how it works.

Reply:
how's about `getResponseContentFromCache` then? the current method makes it sound like the content will be available much more often than it actually is (if we're making wishes I'd also want to have this `| undefined` like our DOM utilities do when this thing throws).